### PR TITLE
[10.0] [ADD] sale_procurement_group_by_requested_date

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**
Supersedes #541

### CODE REVIEW

**Comment by AaronHForgeFlow** [comment]
Needs fixing

**Comment by AaronHForgeFlow** [comment]
ping @mreficent

**Comment by MiquelRForgeFlow** [comment]
If removing dependency, I think this needs a glue module.

**Comment by AaronHForgeFlow** [comment]
sale_sourced_by_line is deprecated as long as we can add routes to the sale order lines, IMO no glue module is needed.

File: sale_procurement_group_by_requested_date/__manifest__.py
Comment (nikul-serpentcs) [inline sale_procurement_group_by_requested_date/__manifest__.py]: Replace URL as per [this](https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L11)

File: sale_procurement_group_by_requested_date/models/sale_order.py
Comment (nikul-serpentcs) [inline sale_procurement_group_by_requested_date/models/sale_order.py]: @aheficent no need, Remove it, Because already done [here](https://github.com/OCA/OCB/blob/10.0/addons/sale/models/sale.py#L638)

File: sale_procurement_group_by_requested_date/tests/test_sale_procurement_group_by_requested_date.py
Comment (nikul-serpentcs) [inline sale_procurement_group_by_requested_date/tests/test_sale_procurement_group_by_requested_date.py]: `self.env['procurement.group']` & `self.env['procurement.order']` use multiple time, add like global variable.

**Review by nikul-serpentcs — COMMENTED** [review]
Minor Change

**Comment by AaronHForgeFlow** [comment]
@nikul-serpentcs comments attended

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM

please squash

**Review by nikul-serpentcs — APPROVED** [review]
Code Review LGTM :+1:

**Comment by AaronHForgeFlow** [comment]
It seems the tests for `sale_mrp_link` are failing here. I will check if it is due this module

**Comment by MiquelRForgeFlow** [comment]
Have you checked?

**Comment by AaronHForgeFlow** [comment]
> Have you checked?

Not yet

**Comment by AaronHForgeFlow** [comment]
Could not reproduce on local. Rebasing.

**Comment by AaronHForgeFlow** [comment]
Glue module is needed, will be proposed here.

**Comment by AaronHForgeFlow** [comment]
Of course the solution was to exclude this module to be installed together with sale_mrp_link. It happened the same to other modules that group procurements.
BTW it seems 10.0 branch is broken anyway. Edit: no, it is not. :smile:

**Comment by tafaRU** [comment]
@aheficent, I think at the moment it's the only solution. See https://github.com/OCA/sale-workflow/issues/717#issuecomment-430231334 as further info.

**Comment by AaronHForgeFlow** [comment]
@tafaRU Thanks, yes I think that is the only solution.

**Comment by AaronHForgeFlow** [comment]
Added glue module for sale_sourced_by_line and sale_procurement_group_by_requested_date.  Sorry for the super large-ugly name.

**Comment by AaronHForgeFlow** [comment]
It seems sale_delivery_split_date has higher priority key than the other modules that group procurements. As having all them installed together makes no sense, I exclude them to be installed together. Only sale_sourced_by_line and sale_procurement_group_by_requested_date is compatible using the glue module proposed here.

**Comment by AaronHForgeFlow** [comment]
I seems there is a test failing on sale_triple_discount module. It is already happening in OCA/10.0 branch for this repo.

**Review by bjeficent — APPROVED** [review]
LGTM :+1:

**Comment by AaronHForgeFlow** [comment]
This can be merged IMO

**Comment by rafaelbn** [comment]
Please @aheficent This branch has conflicts that must be resolved

After this I will merge! Thanks

**Comment by MiquelRForgeFlow** [comment]
@rafaelbn ready to merge. The travis red part is unrelated to this PR, is related to sale_triple_discount that is being fixed in another PR.

**Review by rafaelbn — APPROVED** [review]

**Pull request merged**
